pkg/stub/imagestreams.go

```diff
@@ -473,114 +350,36 @@ func (h *Handler) processImportStatus(is *imagev1.ImageStream, cfg *v1.Config) (
 
 					}
 
-					// if failed previously, but the message has changed, update it
-					if !strings.Contains(importError.Message, message) {
-						msg := h.getImporErrorMessage(is.Name, importError)
-						importError.Message = strings.Replace(importError.Message, msg, h.buildImageStreamErrorMessage(is.Name, message), -1)
-						anyConditionUpdate = true
-					}
-
-					util.ConditionUpdate(cfg, importError)
+				} else {
+					h.clearImageStreamTagError(is.Name, statusTag.Tag)
 				}
 			}
 		}
 
-		/*if !anyErrors {
-			// it is possible after an upgrade that tags can be removed because of EOL processing;
-			// since we do not delete those EOL images from the imagestream status (so as to not break
-			// existing deployments referencing specific tags), it is possible that a valid configuration
-			// has less tags in the spec than the status (vs. an equal amount being the only valid combination)
-			if len(is.Spec.Tags) <= len(is.Status.Tags) {
-				for _, specTag := range is.Spec.Tags {
-					matched := false
-					foundTagInStatus := false
-					foundItemsInStatusTag := false
-					for _, statusTag := range is.Status.Tags {
-						logrus.Debugf("checking spec tag %s against status tag %s with num items %d", specTag.Name, statusTag.Tag, len(statusTag.Items))
-						if specTag.Name == statusTag.Tag {
-							foundTagInStatus = true
-							// if the latest gens have no errors, see if we got gen match
-							if statusTag.Items != nil {
-								for _, event := range statusTag.Items {
-									foundItemsInStatusTag = true
-									if specTag.Generation != nil {
-										logrus.Debugf("checking status tag %d against spec tag %d", event.Generation, *specTag.Generation)
-									}
-									if specTag.Generation != nil &&
-										*specTag.Generation <= event.Generation {
-										logrus.Debugf("got match")
-										matched = true
-										break
-									}
-									nonMatchDetail = fmt.Sprintf("spec tag %s is at generation %d, but status tag %s is at generation %d", specTag.Name, *specTag.Generation, statusTag.Tag, event.Generation)
-								}
-							}
-						}
-					}
-					if !matched {
-						pending = true
-						if len(nonMatchDetail) == 0 {
-							switch {
-							case foundTagInStatus == false:
-								nonMatchDetail = fmt.Sprintf("spec tag %s did not have a matching status tag", specTag.Name)
-							case foundItemsInStatusTag == false:
-								nonMatchDetail = fmt.Sprintf("status tag %s had no items in it", specTag.Name)
-							default:
-								nonMatchDetail = fmt.Sprintf("match failed for some unknown reason; spec: %#v status: %#v", is.Spec.Tags, is.Status.Tags)
-							}
-						}
-						break
-					}
-				}
-				if len(is.Spec.Tags) == 0 {
-					nonMatchDetail = fmt.Sprintf("there were no spec tags")
-				}
-			} else {
-				pending = true
-				nonMatchDetail = "the number of status tags is less than the number of spec tags"
-			}
-		}*/
+		if !anyErrors {
+			h.clearImageStreamErrorLabel(is.Name)
```

Comment:
Here you remove the label, but later on you delete the ConfigMap. Why not delete the ConfigMap here?

Reply:
yeah the delete at 370 will handle it anyway ... I should just remove these 3 lines ... I'll go that way ... and if there are no mre calls to clear imagestream label will delete the method